services/web/server/src/simcore_service_webserver/projects/_db_utils.py

```diff
@@ -370,3 +377,98 @@ async def _get_project(
             project["tags"] = tags
 
         return project
+
+
+def update_workbench(old_project: ProjectDict, new_project: ProjectDict) -> ProjectDict:
+    """any non set entry in the new workbench is taken from the old one if available
+
+    Raises:
+        ProjectInvalidUsageError: it is not allowed to add/remove nodes
+
+    Returns:
+        udpated project
+    """
+
+    old_workbench: dict[NodeIDStr, Any] = old_project["workbench"]
+    updated_project = deepcopy(new_project)
+    new_workbench: dict[NodeIDStr, Any] = updated_project["workbench"]
+
+    if old_workbench.keys() != new_workbench.keys():
+        # it is forbidden to add/remove nodes here
+        raise ProjectInvalidUsageError
+    for node_key, node in new_workbench.items():
+        old_node = old_workbench.get(node_key)
+        if not old_node:
+            continue
+        for prop in old_node:
+            # check if the key is missing in the new node
+            if prop not in node:
+                # use the old value
+                node[prop] = old_node[prop]
```

Comment:
Suggestion: You could probably do something like
```python
for prop in set(old_node.keys())-set(node.keys()):
    node[prop] = old_node[prop]
```
to shorten the inner loop a bit.

Reply:
good point, that was some old code I moved. will give it a try thanks!

Reply:
I will not change it I do not want to break anything right now with this.

Reply:
sure, I understand